pkgs/by-name/sp/spotube/package.nix

```diff
@@ -79,8 +78,13 @@ let
     inherit pname version meta;
 
     src = fetchArtifact {
-      filename = "Spotube-linux-x86_64.deb";
-      hash = "sha256-x75ie9FXunClMT+YZVFlvl2VSDl933QYMRpEYjJ8YhY=";
+      filename =
```

Comment:
What you could also do is the following:

```nix
src = let
  sources = {
    "aarch64-linux" = {
      filename = "Spotube-linux-aarch64.deb";
      hash = "sha256-KBuUAgUU6c/rBkkbpYjSarzckIoi+gRtCkumvtaoras=";
    };
    "x86_64-linux" = {
      filename = "Spotube-linux-x86_64.deb";
      hash = "sha256-vzzK3csyKYP6fKKIoysziBsc2tqrg5LXS/6KoXBtNVI=";
    };
  };

  source = sources.${stdenv.hostPlatform.system};

in fetchArtifact {
  inherit (source) filename hash;
};
```

Reply:
I agree, that's a bit more readable.

An alternative would be something like this:
```nix
src = let
  sources = {
    "aarch64-linux" = fetchArtifact {
      filename = "Spotube-linux-aarch64.deb";
      hash = "sha256-KBuUAgUU6c/rBkkbpYjSarzckIoi+gRtCkumvtaoras=";
    };
    "x86_64-linux" = fetchArtifact {
      filename = "Spotube-linux-x86_64.deb";
      hash = "sha256-vzzK3csyKYP6fKKIoysziBsc2tqrg5LXS/6KoXBtNVI=";
    };
  };
in sources.${stdenv.hostPlatform.system};
```

Reply:
Indeed, even better :)

Reply:
Thanks for the feedback, I applied some changes in the file following this way.